cabal-testsuite/src/Test/Cabal/Script.hs

```diff
@@ -77,14 +77,19 @@ runghc senv mb_cwd env_overrides script_path args = do
 -- script with 'runghc'.
 runnerCommand :: ScriptEnv -> Maybe FilePath -> [(String, Maybe String)]
               -> FilePath -> [String] -> IO (FilePath, [String])
-runnerCommand senv _mb_cwd _env_overrides script_path args = do
+runnerCommand senv mb_cwd _env_overrides script_path args = do
     (prog, _) <- requireProgram verbosity runghcProgram (runnerProgramDb senv)
     return (programPath prog,
             runghc_args ++ ["--"] ++ map ("--ghc-arg="++) ghc_args ++ [script_path] ++ args)
   where
     verbosity = runnerVerbosity senv
     runghc_args = []
-    ghc_args = runnerGhcArgs senv
+    ghc_args = runnerGhcArgs senv ++ srcArg
```

Comment:
I think the more precise way to achieve this is to pass cwd to `runnerGhcArgs` and modify `ghcOptSourcePath`.

I do note though that there `ghcOptSourcePathClear` is set, which seems at odds with this change.. what is `mb_cwd` typically set to?

Reply:
It's typically set to the path to the test in the test-suite, see e.g. this code in `Test.Cabal.Prelude.setup''`:

```haskell
  do r <- liftIO $ runghc (testScriptEnv env)
                          (Just (testCurrentDir env))
                          (testEnvironment env)
                          (testCurrentDir env </> prefix </> "Setup.hs")
                          (NE.toList full_args)
 ```